### PR TITLE
Wilderness lines world map overlay

### PR DESCRIPTION
**Pull request by raiyni**
Still need to add tooltip information and proper comments

### CODE REVIEW

File: runelite-api/src/main/java/net/runelite/api/Line.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+	public Line(int[] start, int[] end)
+	{
+		this.start = start;
+		this.end = end;
+	}
+
+	public Line()
+	{
+		this.start = new int[2];
```
Comment (Nightfirecat) [inline runelite-api/src/main/java/net/runelite/api/Line.java]: This should be a `Point`, not `int[2]`.

File: runelite-api/src/main/java/net/runelite/api/Line.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+		this.end = new int[2];
```
Comment (Nightfirecat) [inline runelite-api/src/main/java/net/runelite/api/Line.java]: Should also be a `Point`.

File: runelite-api/src/main/java/net/runelite/api/Line.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+	}
+
+
+	public int[] getEnd()
+	{
+		return end;
+	}
+
+	@Override
+	public boolean equals(Object obj)
```
Comment (Nightfirecat) [inline runelite-api/src/main/java/net/runelite/api/Line.java]: I believe `@EqualsAndHashCode` on the class should implement this?

File: runelite-api/src/main/java/net/runelite/api/Line.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+	}
+
+	@Override
+	public String toString()
+	{
+		return "Line{" + "start=" + start[0] + "," + start[1] + " - end=" + end[0] + "," + end[1] + '}';
+	}
+
+
+	public int[] getStart()
```
Comment (Nightfirecat) [inline runelite-api/src/main/java/net/runelite/api/Line.java]: Use `@Getter` and `@Setter`.

File: runelite-api/src/main/java/net/runelite/api/coords/WorldLine.java [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+ * WorldPoints are immutable. Methods that modify the properties create a new
+ * instance.
+ */
+@Value
+public class WorldLine
+{
+	/**
+	 * start coordinate.
+	 */
+	private final int[] start;
```
Comment (Nightfirecat) [inline runelite-api/src/main/java/net/runelite/api/coords/WorldLine.java]: Use `Point`.

File: runelite-api/src/main/java/net/runelite/api/coords/WorldLine.java [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+		return new WorldLine(start, end, plane + dz);
+	}
+
+
+	/**
+	 * Gets the ID of the region containing this tile.
+	 *
+	 * @return the region ID
+	 */
+	//public int getRegionID()
```
Comment (Nightfirecat) [inline runelite-api/src/main/java/net/runelite/api/coords/WorldLine.java]: Remove commented code.

File: runelite-api/src/main/java/net/runelite/api/coords/WorldLine.java [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+//	{
+//		return ((x >> 6) << 8) | (y >> 6);
+//	}
+	public boolean isValid()
```
Comment (Nightfirecat) [inline runelite-api/src/main/java/net/runelite/api/coords/WorldLine.java]: Can't this be guaranteed by requiring `@Nonnull` on start and end points?

File: runelite-client/src/main/java/net/runelite/client/plugins/worldmap/WildernessLinesData.java [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+package net.runelite.client.plugins.worldmap;
+
+import java.awt.Color;
+import lombok.Getter;
+import net.runelite.api.coords.WorldLine;
+
+@Getter
+public enum WildernessLinesData
+{
+	WILDERNESS_5(5, new WorldLine(new int[]{2945, 3552}, new int[]{3383, 3552}, 0), Color.CYAN),
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/worldmap/WildernessLinesData.java]: If they are all using the same color, don't require it as a parameter.

**Comment by raiyni** [comment]
tooltips added

**Comment by ShaunDreclin** [comment]
The cyan lines are a bit jarring, maybe a shade of grey would be nicer?

**Comment by Nightfirecat** [comment]
It could be a configuration, I suppose. :woman_shrugging:

**Comment by deathbeam** [comment]
I agree that the color should be changed to something less debug-like, its quite to hard to look at that atm.

**Comment by raiyni** [comment]
Added color config, dungeon lines, and decreased clickbox padding to mouseover icons

**Pull request closed without merging**